cas-server-webapp-cookie/src/main/java/org/jasig/cas/web/support/DefaultCasCookieValueManager.java

```diff
@@ -53,7 +53,11 @@ public DefaultCasCookieValueManager(@Qualifier("defaultCookieCipherExecutor")
     public String buildCookieValue(final String givenCookieValue, final HttpServletRequest request) {
         final StringBuilder builder = new StringBuilder(givenCookieValue);
 
-        final String remoteAddr = request.getRemoteAddr();
+        String remoteAddr = request.getHeader("X-FORWARDED-FOR");
```

Comment:
Shouldn't we rely on the `ClientInfo` here? Which has exactly the same issue: obtaining the IP from an header or directly from `request.getRemoteAddr`?